src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -880,10 +933,15 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
             return builder.MostCommonValue;
         }
 
-        internal override bool IsNullableAnalysisEnabled()
+        internal sealed override bool IsNullableAnalysisEnabled()
+        {
+            return IsNullableAnalysisEnabled(flags.IsNullableAnalysisEnabled);
+        }
+
+        protected virtual bool IsNullableAnalysisEnabled(bool isNullableAnalysisEnabledFlag)
         {
             Debug.Assert(!this.IsConstructor()); // Constructors should use IsNullableEnabledForConstructorsAndInitializers() instead.
-            return flags.IsNullableAnalysisEnabled;
+            return isNullableAnalysisEnabledFlag;
         }
```

Comment:
this pattern allows for:

1. not exposing `Flags` or `flags` just for the need of one subclass.
2. the ability for the base type to assert this isn't called for a constructor.
3. the ability for the derived type to continue making an abstract-override here so that subtypes of it still have to overrid.e